src/main/java/seedu/address/model/person/Role.java

```diff
@@ -0,0 +1,74 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's role in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidRole(String)}
+ */
+public class Role {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Roles should be either 'STUDENT', 'TA', or 'PROFESSOR'.";
+
+    /**
+     * Enum representing different roles in an educational system.
+     * This enum defines three role types: STUDENT, TA (Teaching Assistant), and PROFESSOR.
+     */
+    public enum RoleType {
+        STUDENT, TA, PROFESSOR
+    }
+
+    public final RoleType role;
+
+    /**
+     * Constructs a {@code role}.
+     *
+     * @param role A valid role.
```

Comment:
Feature-design-wise, requiring the **user input** to be in uppercase is highly undesirable.

* It violates the constraints-typing-preferred --- typing uppercase letters take more time than typing lowercase letters. (Actually, it may be a good idea to allow abbreviation to just `s`, `t` or `p` (optional).
* Besides, the docstring is unclear (what roles are valid?)
*